### PR TITLE
RTS-1050: Add EXPLAIN command

### PR DESCRIPTION
**Pull request by hazen**
Remove existing `explain_query_print` and roll functionality into `explain_query`

Related PRs:
https://github.com/basho/riak_ql/pull/135 
https://github.com/basho/riak_shell/pull/46
https://github.com/basho/riak_test/pull/1118

### CODE REVIEW

File: include/riak_kv_ts.hrl [hunk 1]
```diff
@@ -81,4 +81,9 @@
             'SHOW_TABLES' = <<>>  :: binary()
         }).
 
+-record(riak_sql_explain_query_v1,
```
Comment (andytill) [inline include/riak_kv_ts.hrl:105]: This should use a macroed record name like DDL.

File: src/riak_kv_ts_util.erl [hunk 2]
```diff
@@ -437,69 +432,76 @@ explain_query(QueryString) ->
...
-            StartKey3 = [">  ", StartKey2]
-    end,
-    case lists:keyfind(end_inclusive, 1, EndKey1) of
-        {end_inclusive,true} ->
-            EndKey3 = ["<= ", EndKey2];
-        _ ->
-            EndKey3 = ["<  ", EndKey2]
-    end,
-    {StartKey3, EndKey3, Filter}.
+key_element_to_string(V) when is_binary(V) -> binary_to_list(V);
```
Comment (andytill) [inline src/riak_kv_ts_util.erl:476]: I'm wondering if all this string manipulation stuff should go in the riak_ql to string module that converts DDLs to strings.

File: src/riak_kv_ts_util.erl [hunk 3]
```diff
@@ -28,6 +28,7 @@
          apply_timeseries_bucket_props/3,
          build_sql_record/3,
          encode_typeval_key/1,
+         explain_query/1, explain_query/2,
```
Comment (andytill) [inline src/riak_kv_ts_util.erl:31]: Alphabetically sorted exports ❤️ ❤️ ❤️

File: src/riak_kv_ts_util.erl [hunk 4]
```diff
@@ -437,69 +432,76 @@ explain_query(QueryString) ->
...
-filter_to_string(Filter) ->
-    ["FILTER ", filter_to_string2(Filter)].
+-spec key_to_string({binary(),any(),term()}) -> string().
+key_to_string({Field, _Op, Value}) ->
+    lists:flatten(io_lib:format("~s = ~s",
+        [key_element_to_string(Field), element_to_quoted_string(Value)])).
 
 %%
-filter_to_string2({const,V}) ->
+-spec filter_to_string([]| {const,any()} | {atom(),any(),any()}) ->
```
Comment (andytill) [inline src/riak_kv_ts_util.erl:493]: Does this handle more complicated where clauses.

``` sql
x = 9 AND (v = 'j' OR v = 'r') AND f = true
```

**Comment by andytill** [comment]
Can you link the relevant PRs for this change in this one?

File: include/riak_kv_ts.hrl (hunk 1 above)
Comment (hazen) [inline include/riak_kv_ts.hrl:105]: Do we want that for all the query record types?

**Comment by paegun** [comment]
+1 69a52ddd92cf3e82825e3c1f6e23cff67fd26165

**Comment by hazen** [comment]
@andytill All of the PRs are in the Jira issue: https://bashoeng.atlassian.net/browse/RTS-1050

File: src/riak_kv_ts_util.erl (hunk 4 above)
Comment (hazen) [inline src/riak_kv_ts_util.erl:493]: Great question. Let me try.

**Comment by andytill** [comment]
@javajolt yeah, but just for us philistines that don't check JIRA?

File: src/riak_kv_ts_util.erl (hunk 4 above)
Comment (hazen) [inline src/riak_kv_ts_util.erl:493]: Added to the riak_test test in https://github.com/basho/riak_test/pull/1118

Comment (andytill) [inline src/riak_kv_ts_util.erl:493]: Do you have a commit hash? I can't see it.

File: include/riak_kv_ts.hrl (hunk 1 above)
Comment (hmmr) [inline include/riak_kv_ts.hrl:105]: For the newly created ones, yes.

Comment (andytill) [inline include/riak_kv_ts.hrl:105]: I forgot to mention it in the mumble, but I thought we wanted to do everything **one way**. The DDL has been macroed a while back, I have had to macro the sql_select record in group by, and param in descending keys cc @gordonguthrie

**Comment by paegun** [comment]
+1 14edacc

**Comment by paegun** [comment]
+1 9f5ef9d

**Comment by paegun** [comment]
@borshop merge

**Pull request merged**